### PR TITLE
Fix LR scheduler cooldown

### PR DESCRIPTION
**Pull request by stephenroller**
**Patch description**
Context:
- Originally in ParlAI, fixed LR schedulers like cosine/linear would consume (warmup_updates + max_lr_steps) updates, cooling down to 0 eventually
- #3379 changed this behavior such that only max_lr_steps would be consumed, but did not change the cooldown to be faster
- This PR changes it so the full cooldown is completed by the end of max_lr_steps.

**Testing steps**
Adjusted CI, new assertions.

### CODE REVIEW

**Review by emilydinan — APPROVED** [review]
thanks for the fix! this lgtm

File: parlai/core/torch_agent.py [hunk 1]
```diff
@@ -1055,14 +1065,15 @@ def init_optim(self, params, optim_states=None, saved_optim_type=None) -> bool:
...
+            return True
+
         if optim_states and saved_optim_type != opt['optimizer']:
             # we changed from adam to adamax, or sgd to adam, or similar
             logging.warning('Not loading optim state since optim class changed.')
-            return False
+            return True
         elif optim_states:
             # check for any fp16/fp32 conversions we need to do
             optimstate_fp16 = 'loss_scaler' in optim_states
```
Comment (klshuster) [inline parlai/core/torch_agent.py:1079]: i can't leave a comment for it, but are the semantics correct for line 1099? the `elif not optimstate_fp16 and self.fp16` block? are we returning `True` always because of the lower precision conversion?

File: parlai/nn/lr_scheduler.py [hunk 2]
```diff
@@ -502,13 +507,13 @@ def __init__(
...
-        self.max_lr_steps = max_lr_steps
+        assert self.warmup_updates >= 0
+        self.max_lr_steps = max_lr_steps - self.warmup_updates
         self.scheduler = optim.lr_scheduler.LambdaLR(optimizer, self._linear_lr)
 
     def _linear_lr(self, step):
         # this multiplicative factor ensures linear decay rate
-        # lr_mult = float(self.max_lr_steps - step - 1) / float(self.max_lr_steps - step)
-        lr_mult = max(0.0, 1e-6 + (1.0 - step / self.max_lr_steps) * (1 - 1e-6))
+        lr_mult = max(0.0, 1.0 - step / self.max_lr_steps)
```
Comment (klshuster) [inline parlai/nn/lr_scheduler.py:516]: we dont need `1e-6` anymore?

**Review by klshuster — APPROVED** [review]

File: parlai/nn/lr_scheduler.py (hunk 2 above)
Comment (stephenroller) [reply parlai/nn/lr_scheduler.py:516]: I made an executive call to let it actually go to 0 :P

File: parlai/core/torch_agent.py (hunk 1 above)
Comment (stephenroller) [reply parlai/core/torch_agent.py:1079]: agreed
